### ISSUE TITLE
Cryptic error related to knife integration (affects any knife commands)

### ISSUE THREAD

**Issue by patcon** [applicationsonline/librarian#32]
I wish I could give more on this without giving my whole repo, but I think it somehow has something to do with the cookbooks themselves... I've gone back and forth between two chef-repos that I'm using librarian for, and the only diff in the `knife.rb` is the name of the organization on opscode platform (and the fact that there's a comment on the first line in the working one).

Any thoughts on how to troubleshoot this better? For now, I'll just hardcode `cookbooks` and remove the `require`.

```
ERROR: You have an error in your config file /Users/homebase/new-chef-repo/.chef/knife.rb
NoMethodError: undefined method `to_str' for nil:NilClass
  /Users/homebase/new-chef-repo/.chef/knife.rb:1:in `from_file'

     # /Users/homebase/new-chef-repo/.chef/knife.rb
  1: require 'librarian/chef/integration/knife'
  2: 
  3: current_dir = File.dirname(__FILE__)
  4: 
```

**Comment by patcon** [comment]
OK, so I ran librarian-chef clean, the update/install again, and was thrown an error stating that one of my cookbooks was missing a metadata.rb. Checked it out, and somehow only the templates dir had been copied into its cookbook dir in cookbooks-sources (not even a recipe dir was present).

I fixed this and it seems to have resolved itself. Leaving this open in case it's a case for alternate error handling

Cheers Jay!
